dsa-solutions/lc-solutions/0200-0299/0218-the-skyline-problem.md

```diff
@@ -0,0 +1,232 @@
+                result.append([x, current_max_height])
+                prev_max_height = current_max_height
+        
+        return result
+```
+
+### Java
+```java
+import java.util.*;
+
+class Solution {
+    public List<List<Integer>> getSkyline(int[][] buildings) {
+        List<int[]> events = new ArrayList<>();
+        for (int[] building : buildings) {
+            events.add(new int[] {building[0], -building[2], building[1]}); // start event
+            events.add(new int[] {building[1], 0, 0}); // end event
+        }
+        events.sort((a, b) -> a[0] != b[0] ? a[0] - b[0] : a[1] - b[1]);
+        
+        List<List<Integer>> result = new ArrayList<>();
+        PriorityQueue<int[]> maxHeap = new PriorityQueue<>((a, b) -> b[0] - a[0]);
+        maxHeap.offer(new int[] {0, Integer.MAX_VALUE});
+        int prevMaxHeight = 0;
+        
+        for (int[] event : events) {
+            int x = event[0], negHeight = event[1], right = event[2];
+            while (maxHeap.peek()[1] <= x) {
+                maxHeap.poll();
+            }
+            if (negHeight != 0) {
+                maxHeap.offer(new int[] {-negHeight, right});
+            }
+            int currentMaxHeight = maxHeap.peek()[0];
+            if (prevMaxHeight != currentMaxHeight) {
+                result.add(Arrays.asList(x, currentMaxHeight));
+                prevMaxHeight = currentMaxHeight;
+            }
+        }
+        
+        return result;
+    }
+}
+```
+
+### C
+```c
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct {
+    int x;
+    int h;
+    int isStart;
+} Event;
+
+int cmp(const void *a, const void *b) {
+    Event *e1 = (Event *)a;
+    Event *e2 = (Event *)b;
+    if (e1->x != e2->x) return e1->x - e2->x;
+    if (e1->isStart && e2->isStart) return e2->h - e1->h;
+    if (!e1->isStart && !e2->isStart) return e1->h - e2->h;
+    return e1->isStart ? -1 : 1;
+}
+
+int* getSkyline(int** buildings, int buildingsSize, int* buildingsColSize, int* returnSize) {
+    Event *events = malloc(buildingsSize * 2 * sizeof(Event));
+    for (int i = 0; i < buildingsSize; ++i) {
+        events[i * 2] = (Event){buildings[i][0], buildings[i][2], 1};
+        events[i * 2 + 1] = (Event){buildings[i][1], buildings[i][2], 0};
+    }
+    qsort(events, buildingsSize * 2, sizeof(Event), cmp);
+    
+    int *result = malloc(buildingsSize * 2 * 2 * sizeof(int));
+    int resultSize = 0;
+    
+    int *heights = calloc(buildingsSize * 2, sizeof(int));
+    int heightsSize = 0;
+    int prevMaxHeight = 0;
+    
+    for (int i = 0; i < buildingsSize * 2; ++i) {
+        if (events[i].isStart) {
+            heights[heightsSize++] = events[i].h;
+        } else {
+            for (int j = 0; j < heightsSize; ++j) {
+                if (heights[j] == events[i].h) {
+                    heights[j] = heights[--heightsSize];
+                    break;
+                }
+            }
+        }
+        
+        int currentMaxHeight = 0;
+        for
```

Comment:
write complete code

Reply:
> write complete code

@Ajay-Dhangar sir fixed #3343 and did the correction.